### PR TITLE
Load proper metrics for do_eval tasks

### PR DESCRIPTION
**Pull request by awebson**
@craffel This is a first stab at loading the appropriate metric for each dataset_subset_template.

Caveats:
1. `t5.evaluation.metrics` seems to do something special with SuperGLUE WSC. So I haven't handled that yet.
2. No seqio pre/postprocessing is handled, which is most relevant for…
3. The `rank_classification` metric, which I assume implements the trick of of plugging in the answers and then compare LM probabilities as class predictions.
4. For the caching error of leading space in ground truth targets, stripping of targets is not yet handled by metrics. We talked about using functools.partial but that can only fix an argument to a constant, not applying strip() to an argument. We can write a lambda wrapping every metric, but maybe postprocessing is still the way to go.

### CODE REVIEW

**Comment by craffel** [comment]
The postprocess_fn does get applied to the target: https://github.com/google/seqio/blob/main/seqio/evaluation.py#L209 so, for all tasks, I would suggest adding the equivalent of strip() as a postprocessor.
https://github.com/google/seqio#postprocessor

File: promptsource/seqio_tasks/tasks.py [hunk 1]
```diff
@@ -37,6 +37,18 @@
...
+
+def strip_whitespace(output_or_target, example=None, is_target=False):
+    """
+    Temporary hack. Cached tasks from promptsource all have a leading space
+    on the ground-truth targets.
+    """
+    if is_target:
+        return output_or_target.strip()
+    else:
+        return output_or_target
```
Comment (craffel) [inline promptsource/seqio_tasks/tasks.py:49]: I think it's probably fair/correct to strip() both inputs and targets.. WDYT?

File: promptsource/seqio_tasks/load_annotated_prompts.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            # Required for all prompts with non_trivial_choices_hidden,
+            # but could be used for other tasks as well where answer choices are given.
+            if "metrics" not in task:
+                task["metrics"] = [rank_classification]
+            elif rank_classification not in task["metrics"]:
+                task["metrics"].append(rank_classification)
+
+        # should be already handled by NON_GLUE_METRICS
+        # if task['generative_true_task'] or task['generative_non_true_task']:
+        #     task['metrics'] = rogue
```
Comment (craffel) [inline promptsource/seqio_tasks/load_annotated_prompts.py:90]: I know this is a comment but it's rouge not rogue

File: promptsource/seqio_tasks/tasks.py [hunk 3]
```diff
@@ -32,6 +37,18 @@
     ("tweet_eval", "irony"),
 ]
 
+
+def strip_whitespace(output_or_target, example=None, is_target=False):
+    """
+    Temporary hack. Cached tasks from promptsource all have a leading space
```
Comment (craffel) [inline promptsource/seqio_tasks/tasks.py:43]: TBH I think it's probably not temporary...

File: promptsource/seqio_tasks/load_annotated_prompts.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
+import csv
```
Comment (craffel) [inline promptsource/seqio_tasks/load_annotated_prompts.py:1]: Sort of a nitpick (so feel free to disregard) but I would include these functions in tasks.py and then just treat this as a diagnostic script.

**Review by craffel — CHANGES_REQUESTED** [review]

File: promptsource/seqio_tasks/load_annotated_prompts.py (hunk 4 above)
Comment (awebson) [reply promptsource/seqio_tasks/load_annotated_prompts.py:1]: Sounds good. I will move them in the next commit.

File: promptsource/seqio_tasks/load_annotated_prompts.py (hunk 2 above)
Comment (awebson) [reply promptsource/seqio_tasks/load_annotated_prompts.py:90]: Good catch!

File: promptsource/seqio_tasks/tasks.py (hunk 1 above)
Comment (awebson) [reply promptsource/seqio_tasks/tasks.py:49]: Sure. We can double check later but can definitely strip both to be safe for now.

**Review by awebson — COMMENTED** [review]
Pushed new commits that strip both output and target in postprocessor and fixed typos.

**Comment by craffel** [comment]
Thanks!

**Pull request merged**